core/runtime/src/main/java/org/eclipse/sw360/antenna/workflow/stubs/AbstractComplianceChecker.java

```diff
@@ -148,18 +100,18 @@ private void reportResults(IProcessingReporter reporter, Set<IEvaluationResult>
         }
 
     private void reportSingleResult(IProcessingReporter reporter, IEvaluationResult result) {
-        result.getFailedArtifacts().forEach(a -> reporter.add(MessageType.RULE_ENGINE,
-                result.getId() + " (" + result.getSeverity() + "): "
-                        + getCoordinates(a) + " : " + result.getDescription()));
-    }
-
-    private String getCoordinates(Artifact a) {
-        Optional<Coordinate> coordinate = a.getCoordinateForType(Coordinate.Types.MAVEN);
-        if (! coordinate.isPresent()) {
-            coordinate = a.getCoordinates().stream().findFirst();
+        String resultString = String.format("Policy Violation: %s (%s): %s%n          %s",
+                result.getId(),
+                result.getSeverity(),
+                result.getFailedArtifacts().stream().map(Artifact::artifactAsCoordinate).collect(Collectors.joining(", ")),
+                result.getDescription());
```

Comment:
pretty similar to the string-ification in `AntennaWorkflow`. would it be worth it to make a toString() function in EvaluationResult?

Reply:
I have made a new proposal, which uses a interface default message, but perhaps this is also not the best solution. It enters output related stuff to a pure data interface 🤔